### ISSUE TITLE
Crashing in cleanup

### ISSUE THREAD

**Issue by motoglaf** [opentoonz/opentoonz#5296]
### Description

I have had several different crashes for the mac version of opentoonz when dealing with the cleanup room. Is there any fixes for this? Below is one of my logs. I have been trying to use the latest nightly builds.

### Steps to Reproduce

1. Select the cleanup room and then set clean up preview.
2. Change from grayscale to color under the line processing drop-down.

### Expected Behavior

Should change options without application failing.

### Screenshots, Video & Crash Logs

OpenToonz 1.7.1  (Build Jan 14 2024)

Report Date: 2024-02-11 18:03:19
Crash Reason: (SIGSEGV) Segmentation Fault

Build ABI: x86_64-little_endian-lp64
Operating System: macOS 10.16
OS Kernel: 23.3.0
CPU Threads: 8

GPU Vendor: Intel Inc.
GPU Model: Intel(R) Iris(TM) Plus Graphics 645
GPU Version: 2.1 INTEL-22.1.29

Crash File: /Users/michaelnorris/Library/Application Support/OpenToonz/OpenToonz/crash/Crash-20240211-180319.log

Application Dir: /Applications/OpenToonz.app/Contents/MacOS
Stuff Dir: /Applications/OpenToonz/OpenToonz_stuff

Project Name: Kiwi Cowboy
Scene Name: SC11_SH_1
Project Path: /Applications/OpenToonz/OpenToonz_stuff/projects/Kiwi Cowboy/Kiwi Cowboy_otprj.xml
Scene Path: /Applications/OpenToonz/OpenToonz_stuff/projects/Kiwi Cowboy/scenes/SC11_SH_1.tnz

==== Backtrace ====
  1> 3   ???                                 0x0000000000000000 0x0 + 0
  2> 4   OpenToonz                           0x000000010edd833b _ZN20CleanupPaletteViewer7setModeEb + 27
  3> 5   OpenToonz                           0x000000010ee3502f _ZN19CleanupSettingsPane16updateVisibilityEv + 479
  4> 6   OpenToonz                           0x000000010ee34cb1 _ZN19CleanupSettingsPane9updateGuiEP17CleanupParametersS1_ + 625
  5> 7   OpenToonz                           0x000000010ee346b2 _ZN19CleanupSettingsPane9updateGuiEb + 50
  6> 8   QtCore                              0x00000001148afbac _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3868
  7> 9   OpenToonz                           0x000000010ef552cc _ZN20CleanupSettingsModel12modelChangedEb + 60
  8> 10  OpenToonz                           0x000000010ee3592c _ZN19CleanupSettingsPane23onGenericSettingsChangeEv + 748
  9> 11  QtCore                              0x00000001148afbac _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3868
 10> 12  QtWidgets                           0x0000000112fb8079 _ZN9QComboBox15setCurrentIndexEi + 297
 11> 13  QtWidgets                           0x0000000112fbd087 _ZN9QComboBox11setItemTextEiRK7QString + 279
 12> 14  QtCore                              0x00000001148afbac _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3868
 13> 15  QtWidgets                           0x0000000112fbadf9 _ZN25QComboBoxPrivateContainer11eventFilterEP7QObjectP6QEvent + 1193
 14> 16  QtCore                              0x000000011487da99 _ZN23QCoreApplicationPrivate29sendThroughObjectEventFiltersEP7QObjectP6QEvent + 217
 15> 17  QtWidgets                           0x0000000112ebe518 _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent + 248
 16> 18  QtWidgets                           0x0000000112ec1424 _ZN12QApplication6notifyEP7QObjectP6QEvent + 7428
 17> 19  QtCore                              0x000000011487d7f4 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent + 212
 18> 20  QtWidgets                           0x0000000112ebee57 _ZN19QApplicationPrivate14sendMouseEventEP7QWidgetP11QMouseEventS1_S1_PS1_R8QPointerIS0_Ebb + 903
 19> 21  QtWidgets                           0x0000000112f1680a _ZN14QDesktopWidget11qt_metacallEN11QMetaObject4CallEiPPv + 10186
 20> 22  QtWidgets                           0x0000000112f1523e _ZN14QDesktopWidget11qt_metacallEN11QMetaObject4CallEiPPv + 4606
 21> 23  QtWidgets                           0x0000000112ebe52d _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent + 269
 22> 24  QtWidgets                           0x0000000112ebf8f2 _ZN12QApplication6notifyEP7QObjectP6QEvent + 466
 23> 25  QtCore                              0x000000011487d7f4 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent + 212
 24> 26  QtGui                               0x0000000113a9e811 _ZN22QGuiApplicationPrivate17processMouseEventEPN29QWindowSystemInterfacePrivate10MouseEventE + 3489
 25> 27  QtGui                               0x0000000113a9dfbb _ZN22QGuiApplicationPrivate17processMouseEventEPN29QWindowSystemInterfacePrivate10MouseEventE + 1355
 26> 28  QtGui                               0x0000000113a837eb _ZN22QWindowSystemInterface22sendWindowSystemEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE + 219
 27> 29  libqcocoa.dylib                     0x000000010f685a58 qt_plugin_instance + 199592
 28> 30  CoreFoundation                      0x00007ff817c0dff7 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
 29> 31  CoreFoundation                      0x00007ff817c0df99 __CFRunLoopDoSource0 + 157
 30> 32  CoreFoundation                      0x00007ff817c0dd64 __CFRunLoopDoSources0 + 215
 31> 33  CoreFoundation                      0x00007ff817c0c9e1 __CFRunLoopRun + 919
 32> 34  CoreFoundation                      0x00007ff817c0c082 CFRunLoopRunSpecific + 557
 33> 35  HIToolbox                           0x00007ff8224fe9d9 RunCurrentEventLoopInMode + 292
 34> 36  HIToolbox                           0x00007ff8224fe616 ReceiveNextEventCommon + 201
 35> 37  HIToolbox                           0x00007ff8224fe531 _BlockUntilNextEventMatchingListInModeWithFilter + 66
 36> 38  AppKit                              0x00007ff81b21cc39 _DPSNextEvent + 880
 37>  38> atos cannot load symbols for the file  for architecture x86_64.
 39> 41  libqcocoa.dylib                     0x000000010f68486c qt_plugin_instance + 195004
 40> 42  QtCore                              0x000000011487997f _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE + 431
 41> 43  QtCore                              0x000000011487dd82 _ZN16QCoreApplication4execEv + 130
 42> 44  OpenToonz                           0x000000010ea119a1 main + 15553
 43> 45  dyld                                0x00007ff8177a7386 start + 1942
==== End ====

### OpenToonz Version

Nightly (Specify below)

### OpenToonz Version Information

OpenToonz 1.7.1  (Build Jan 14 2024)

### Operating System

macOS

### GPU

Intel

### Graphics Tablet

None

**Comment by RodneyBaker** [comment]
Perhaps related to other reports where cleanup was being used.  

REF:  #5257

**Comment by motoglaf** [comment]
> Perhaps related to other reports where cleanup was being used.
> 
> REF: #5257

Yeah I see it has happened in a Windows environment too according to that report. I would use GTS but it's not available on macos. Cleanup seems really unstable as sometimes it will work but most of the time after enough time with the application open it crashes. I will see if I can record a video of it soon to show it.

Thanks,

**Comment by motoglaf** [comment]
See recording.

https://github.com/opentoonz/opentoonz/assets/133850411/41f55828-5a8d-45a2-b15b-f8d0ea598ea4

**Comment by RodneyBaker** [comment]
There was someone else that had this problem and... if I recall correctly... they addressed it by changing their Cleanup Settings.
I don't remember the specifics of that although in your video my brain fired on seeing the X and Y settings be set to zero.  I doubt that is it but look closely at all of the settings and see if changing them doesn't make a difference.  If it is the zeros then perhaps some divide by zero calculations leading to crash.  That's not likely what is the problem but... some calculation apparently is...

**Comment by artisteacher** [comment]
https://github.com/opentoonz/opentoonz/issues/654

This has been an issue as long as I've used OT. Which is why I don't try to use Cleanup anymore.

**Comment by RodneyBaker** [comment]
This issue is likely fixed in current nightly releases.

**Comment by RodneyBaker** [comment]
Recommend retesting and confirmation in light of recent changes/fixes.
